engine/src/main/java/org/terasology/logic/console/commands/CoreCommands.java

```diff
@@ -527,7 +527,7 @@ public String spawnPrefab(@Sender EntityRef sender, @CommandParam("prefabId") St
 
         Optional<Prefab> prefab = Assets.getPrefab(prefabName);
         if (prefab.isPresent() && prefab.get().getComponent(LocationComponent.class) != null) {
```

Comment:
Do you have an idea how to use `Optional` in a more idiomatic way? I think the logic can be expressed as follows, but I'm not sure how to end up with the status strings as before...

```java
Assets.getPrefab(prefabName)
  .filter(prefab -> prefab.getComponent(LocationComponent.class) != null)
  .ifPresent(prefab -> entityManager.create(prefab.get(), spawnPos);
```

Reply:
Hmm, I think I would do:
```    return Assets.getPrefab(prefabName).map(prefab -> {
            LocationComponent loc = prefab.getComponent(LocationComponent.class);
            if (loc != null) {
                entityManager.create(prefab, spawnPos);
                return "Done";
            } else {
                return "Prefab cannot be spawned (no location component)";
            }
        }).orElse("Unknown prefab");````

Reply:
That would work... I'm always hesitant to have side effects in `map` and try to use `ifPresent` (`forEach` in Scala) for that purpose. 
I think your proposal is better than using the combination of `isPresent` and `get()` (multiple times). 👍